### PR TITLE
client: do not crash on unknown profile

### PR DESCRIPTION
**Pull request by rhyw**
Fixes OSH-145

### CODE REVIEW

File: osh/client/commands/cmd_version_diff_build.py [hunk 1]
```diff
@@ -198,7 +199,10 @@ def run(self, *args, **kwargs):  # noqa: C901
                 self.parser.error(str(ex))
             options['analyzers'] = analyzers
         if profile:
-            options['profile'] = profile
+            result = verify_scan_profile_exists(self.hub, profile)
+            if result is not None:
+                self.parser.error(result)
+        options['profile'] = profile
```
Comment (lzaoral) [inline osh/client/commands/cmd_version_diff_build.py:205]: Same as above.

File: osh/client/commands/shortcuts.py [hunk 2]
```diff
@@ -145,3 +145,12 @@ def upload_file(hub, srpm, target_dir, parser):
         return hub.upload_file(os.path.expanduser(srpm), target_dir)
     except Fault as e:
         handle_perm_denied(e, parser)
+
+
+def verify_scan_profile_exists(hub, profile_name):
+    """Verify if scan profile exists"""
+    profiles = hub.scan.list_profiles()
+    profile_exists = any(p["name"] == profile_name for p in profiles)
+    if profile_exists:
```
Comment (lzaoral) [inline osh/client/commands/shortcuts.py:154]: There is no need to introduce a local variable that will be set and read only once.
```suggestion
    if any(p["name"] == profile_name for p in profiles):
```

**Review by lzaoral — CHANGES_REQUESTED** [review]
Thank you for your contribution!  See comments below.

Please, amend the commit messages to contain the following string:
```
Resolves: <fill link to OSH-145 Jira ticket>
```

**Comment by rhyw** [comment]
> Thank you for your contribution! See comments below.
> 
> Please, amend the commit messages to contain the following string:
> 
> ```
> Resolves: <fill link to OSH-145 Jira ticket>
> ```

I've been working on other RH internal projects as well. TBH I'm not quite comfortable pasting internal links which were not accessible from outside of the world.

File: osh/client/commands/shortcuts.py (hunk 2 above)
Comment (rhyw) [reply osh/client/commands/shortcuts.py:154]: I originally also want a one liner. but some may say one liner ruins readability.

Anyway, I'm ok with getting rid of this accidentally introduced local vars. Thanks for the suggestion.

File: osh/client/commands/cmd_version_diff_build.py (hunk 1 above)
Comment (rhyw) [reply osh/client/commands/cmd_version_diff_build.py]: Fixed.

**Comment by rhyw** [comment]
> Resolves: <fill link to OSH-145 Jira ticket>

We do have conventions in historical commits. I've amended the commit message to follow that! Thanks for the suggestion.

**Review by lzaoral — CHANGES_REQUESTED** [review]
Perfect, thank you!  I see that the coverage decreased a bit.  Could you please a modify the first `mock-build` task in `containers/scripts/generate_integration_test_coverage.sh` to include the `--profile=default` option?  Looking at the codecov logs, we not have this option covered yet.  The behaviour should be the same because `default` is the implicit profile used by `osh-cli`.

**Comment by rhyw** [comment]
Updated as per suggestion.

Code coupled too much with kobo is the biggest challenge to me. have to traverse repo back and forth to get an idea how different modules work together.

**Review by lzaoral — APPROVED** [review]
LGTM!  Thank you, @rhyw!

**Pull request merged**